test/integration-new/README.md

```diff
@@ -60,4 +60,237 @@ Currently the package is named as `integration-new` because we already have `int
+
+### Adding new test folder
+
+Say for instance, The cni test and suite files in the cni folder has functionality related to CNI component in VPC CNI as you would expect. If you want to add a test that does not belong to any of the modules in the integration-new folder, you will have to create a new folder structure as below
+- ```integration-new``` 
+  - ```new_component_test_xyz``` 
+       - ```new_component_test_xyz/new_component_xyz_suite_test.go```)
+       - ```new_component_test_xyz/xyz_test_1.go``` 
+       - ```new_component_test_xyz/xyz_test_2.go```) 
+       - ```...```
+  - ```cni``` 
+  - ```...```
+
+### Structure of test suite 
+
+#### Logic Components
+
+- ```BeforeSuite``` : All common steps that should be performed before the suite are added here.
+- ```AfterSuite``` : All common steps that should be performed after the suite are added here.
+
+Most of the test suites follow the following structure. Some of the functions used below are just an example for illustration of functionality. Additions or deletions to the below snippet may be required for any new test suite.
+
+```go
+package <package_name>
+
+import (
+        // fmt is imported for printing
+	"fmt"
+        // testing is imported as it is the original go testing module used by ginkgo
+	"testing"
+
+        // The below folders are similar to the ones discussed above
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework" 
+	k8sUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+
+        //ginkgo and the assertion library: gomega are imported below
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+ 
+        //v1 imported for Node libraries
+	v1 "k8s.io/api/core/v1"
+	
+        ...
+)
+
+//Global variables for the suite are defined here
+var f *framework.Framework
+var primaryNode v1.Node
+...
+
+//The function below is the starter function for running tests 
+//for each suite and attaching a fail handler for the same
+func TestSample(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Sample Suite")
+}
+
+//The following function has checks and setup needed before running the suite.
+var _ = BeforeSuite(func() {
+
+	f = framework.New(framework.GlobalOptions)
+         ...
+         
+        // The Sequence of By and Expect are provided by the omega package and 
+        // ensure the correct functionality by providing assertions at every step 
+	input := ...
+
+	By("getting the output for xyz using input ")
+	output, err := func_xyz(input)
+	Expect(err).ToNot(HaveOccurred())
+ 
+        ...
+  
+
+})
+
+//The following function has checks and setup needed after running the suite.
+var _ = AfterSuite(func() {
+
+        ...
+
+})
+
+```
+### Structure of tests corresponding to each suite
+
+
+#### Logic Components
```

Comment:
This section is ginkgo specific, we can add them as short comments when you explain the example code snippet below. Instead provide a section of troubleshooting tests . Like Checking for cleaned up resources, deployments stuck, running in focussed mode, checking logs, similar to this : https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/docs/guide/troubleshooting.md. We can also have a separate file for troubleshooting.md